locking/admin.py

```diff
@@ -116,22 +117,46 @@ def get_urls(self):
         ]
         return locking_urls + urls
 
+    def get_api_url(self, object_id):
+        app_label, model_name = self._model_info
+
+        reverse_kwargs = {
+            'app': app_label,
+            'model': model_name,
+        }
+        if object_id is not None:
+            reverse_kwargs['object_id'] = object_id
+
+        return reverse('locking:locking-api', kwargs=reverse_kwargs)
```

Comment:
Yes, you're right, Django 1.8 will get `NoReverseMatch` error, if they don't provide proper namespace. For more info, see my comment for app_name.

I discarded my changes for urls and namespace for now.